tests/e2e/e2e_test.go

```diff
@@ -270,108 +276,105 @@ var _ = Describe("odoe2e", func() {
+				})
 
-			It("should be able to delete an application", func() {
-				// Cleanup
-				runCmdShouldPass("odo app delete " + appTestName + "-2 -f")
 			})
 
-			It("should be able to set an application as current", func() {
-				appName := runCmdShouldPass("odo app set " + appTestName)
-				Expect(appName).To(ContainSubstring(appTestName))
-			})
+			// TODO: Check if the application with the same name can be created
 		})
-
-		// TODO: Check if the application with the same name can be created
-	})
+	*/
 
 	Context("should list applications in other project", func() {
 		newProjName := strings.Replace(projName, "odo", "odo2", -1)
 		It("should create a new project", func() {
-			session := runCmdShouldPass("odo project create " + newProjName)
-			Expect(session).To(ContainSubstring(newProjName))
-		})
-
-		It("should get the project", func() {
-			getProj := runCmdShouldPass("odo project get --short")
-			Expect(strings.TrimSpace(getProj)).To(Equal(newProjName))
+			runCmdShouldPass("oc new-project " + newProjName)
+			//session := runCmdShouldPass("odo project create " + newProjName)
+			//Expect(session).To(ContainSubstring(newProjName))
 		})
 
 		It("should show nice message when there is no application in project", func() {
-			appNames := runCmdShouldPass("odo app list")
+			appNames := runCmdShouldPass("odo app list --project " + newProjName)
 			Expect(strings.TrimSpace(appNames)).To(
 				Equal("There are no applications deployed in the project '" + newProjName + "'."))
 		})
 
-		It("should be able to list applications in other project", func() {
-			appNames := runCmdShouldPass("odo app list --project " + projName)
-			Expect(appNames).To(ContainSubstring(appTestName))
-		})
-
-		It("should set the other project as active", func() {
-			setProj := runCmdShouldPass("odo project set --short " + projName)
-			Expect(strings.TrimSpace(setProj)).To(Equal(projName))
+		It("should be able to create a nodejs component with application created", func() {
+			runCmdShouldPass("odo create php testcmp --app e2e-xyzk --git " + testPHPGitURL)
+			runCmdShouldPass("odo push")
 		})
 
-		It("should be able to set an application as current", func() {
-			appName := runCmdShouldPass("odo app set " + appTestName)
-			Expect(appName).To(ContainSubstring(appTestName))
+		It("should be able to list applications in other project", func() {
+			appNames := runCmdShouldPass("odo app list --project " + newProjName)
+			Expect(appNames).To(ContainSubstring(appTestName))
 		})
 	})
 
 	Describe("creating a component", func() {
 		Context("when application exists", func() {
-			var autoGenNodeJSCompName string
+			//var autoGenNodeJSCompName string
 
 			It("should be able to create new imagestream and find it in catalog list", func() {
 				curProj = runCmdShouldPass("oc project -q")
 				curProj = strings.TrimSuffix(curProj, "\n")
 				cmd := fmt.Sprintf("oc create -f "+testNamespacedImage+" -n %s", curProj)
 				runCmdShouldPass(cmd)
+				time.Sleep(5)
```

Comment:
Noooo, no sleeps, please. 
If it is really necessary and this is workaround for some race condition create an issue and add it to the comment

Reply:
Aah yeah I added it for debugging I'll remove it